src/components/grid/grid-container/grid-container.component.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import GridContainerStyle from './grid-container.style';
+
+const GridContainer = (props) => {
+  const {
+    children
+  } = props;
+  return (
+    <GridContainerStyle>
+      {children}
+    </GridContainerStyle>
+  );
+};
+
+const createGridItemPropType = () => {
+  return (props, propName, componentName) => {
+    let error;
+    const prop = props[propName];
+
+    if (!prop) {
+      error = new Error(`\`${componentName}\` requires children of type GridItem.`);
+    } else {
+      React.Children.forEach(prop, (child) => {
+        if (child.type.name !== 'GridItem') {
+          error = new Error(`\`${componentName}\` only accepts children of type GridItem.`);
+        }
+      });
+    }
+    return error;
+  };
+};
+
+const gridItemType = createGridItemPropType();
+
+GridContainer.propTypes = {
+  /** Defines the Components to be rendered within the GridContainer. Requires a GridItem */
+  children: gridItemType
```

Comment:
I recently used that to ensure type. Unless we need that specific error messages you can try to use this (shorter) approach:

```
children: PropTypes.shape({
    type: PropTypes.oneOf([Icon, PropTypes.element])
  }).isRequired
```